File: internal/model/character.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type CharacterModel struct {
	ID              uuid.UUID
	UserID          uuid.UUID
	Name            string
	Class           string
	Level           int
	Experience      int
	HP              int
	Attack          int
	Defense         int
	AttributePoints int
	RankingPoints   int
	CriticalChance  int
	CreatedAt       time.Time
	UpdatedAt       time.Time
}

File: frontend/src/types/character.ts
export interface Character{
    id: string;
    user_id: string;
    name: string;
    class: string;
    level: number;
    experience: number;
    hp: number;
    attack: number;
    defense: number;
    attribute_points: number;
    ranking_points: number;
    critical_chance: number;
    created_at: string;
    updated_at: string;
}

export interface CharacterListResponse {
    characters: Character[];
    total: number;
}

export interface CreateCharacterRequest {
    name: string;
    class: string;
}

export interface UpdateCharacterRequest {
    name: string;
}

export interface AddExperienceRequest {
    experience: number;
}

export interface SpendAttributeRequest {
    hp?: number;
    attack?: number;
    defense?: number;
    critical_chance?: number;
}